Derpy/Utils/Tumblr/TumblrClient.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Sentry;
+using Sentry.Protocol;
+using Serilog;
+
+namespace Derpy.Utils.Tumblr
+{
+    public class TumblrClient : ITumblrClient
+    {
+        private readonly IWebClient _webClient;
+        private readonly string _apiKey;
+
+        private static Uri apiUrl = new Uri("https://api.tumblr.com/v2/blog/");
+        private static JsonSerializerOptions _serializerOptions = new JsonSerializerOptions
+        {
+            PropertyNamingPolicy = new SnakeCaseNamingPolicy()
+        };
+
+        public TumblrClient(IWebClient webClient, IKeyProvider keyProvider)
+        {
+            _webClient = webClient;
+            _apiKey = keyProvider.TumblrApiKey;
+        }
+
+        // https://www.tumblr.com/docs/en/api/v2#posts--retrieve-published-posts
+        // https://github.com/tumblr/docs/blob/master/api.md#posts--retrieve-published-posts
+        public async Task<string[]> GetAllPostUrlsAsync(string blogIdentifier, string tag = null)
+        {
+            var @params = new Dictionary<string, string>
```

Comment:
Why the `@`?

Reply:
`params` is a reserved keyword, and C# allows you to use reserved keywords as parameter names by prefacing it with an `@`. I was basically lazy and couldn't think of a better name for the URL parameter variable, haha!